### ISSUE TITLE
Customize "Go To Declarations or Usages" Shortcut Independent of Multi-Cursor Modifier in Keymap

### ISSUE THREAD

**Issue by mhkafadar** [zed-industries/zed#10007]
### Check for existing issues

- [X] Completed

### Describe the feature

While exploring Zed Editor, I discovered that the "Go To Declarations or Usages" action can be triggered with CMD+Click. However, when attempting to customize this to "Option + Click", the editor doesn't allow such modification in the keymap settings. Additionally, modifying "multi_cursor_modifier": "cmd" to "options" changes the multi-cursor behavior to use the Option key, which is not my intent. I would like to maintain CMD for multi-cursor actions while also setting "Go To Declarations or Usages" to "Option + Click".

### If applicable, add mockups / screenshots to help present your vision of the feature

<img width="299" alt="Screenshot 2024-03-31 at 16 20 52" src="https://github.com/zed-industries/zed/assets/43652059/40023dce-86d0-4635-999c-c499e087bc19">

<img width="176" alt="Screenshot 2024-03-31 at 16 21 16" src="https://github.com/zed-industries/zed/assets/43652059/7616ed1b-e45d-4280-9716-6999f7f22069">

**Comment by wottpal** [comment]
Yeah, I'm looking for the exact same behavior. Could you find something to fix this @Moshyfawn?

**Comment by maxdeviant** [comment]
I think this seems like a reasonable thing to support
